### PR TITLE
deps: update k8s to v0.21.0

### PR DESCRIPTION
**Pull request by klihub**
This should fix recently reported/emerged build errors when building first time from scratch.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -18,43 +20,48 @@ require (
...
+	k8s.io/apimachinery v0.21.0
+	k8s.io/client-go v0.21.0
+	k8s.io/cri-api v0.21.0
+	k8s.io/klog/v2 v2.8.0
+	k8s.io/kubernetes v1.21.0
 	sigs.k8s.io/yaml v1.2.0
 )
 
 replace (
+	github.com/containerd/nri => github.com/klihub/nri v0.0.0-20210916151032-5665bdaef70c
```
Comment (askervin) [inline go.mod:38]: NRI replace shouldn't probably be in this commit.

Comment (marquiz) [reply go.mod:38]: Yeah, otherwise lgtm

Comment (klihub) [reply go.mod]: Thanks fixed that one.

Now it looks to me like `apimachinery v0.22.1` (`goresctrl` commit [`6d75e171`](https://github.com/intel/goresctrl/commit/6d75e171db261ba93da99061108bafb2282c43c2)) causes an inherently incompilable k8s dependency set to get pulled in. I could not find any other fix than downgrade that dependency to not later than `v0.21.0`. AFAICT is the latest tagged release for which all the necessary k8s subpackages split out from `staging` have been properly updated. A real pain in the butt...

Here is a [pending PR](https://github.com/intel/goresctrl/pull/78) doing just that.

File: go.mod [hunk 2]
```diff
@@ -23,40 +23,43 @@ require (
...
-	k8s.io/kube-aggregator v0.0.0 => k8s.io/kube-aggregator v0.0.0-20201113173433-ce05bf9f9509
-	k8s.io/kube-controller-manager v0.0.0 => k8s.io/kube-controller-manager v0.0.0-20201113181424-f91ce16d8de8
-	k8s.io/kube-proxy v0.0.0 => k8s.io/kube-proxy v0.0.0-20201113180133-d8e6aa5a8916
-	k8s.io/kube-scheduler v0.0.0 => k8s.io/kube-scheduler v0.0.0-20201113180716-8fee79fb9051
-	k8s.io/kubectl v0.0.0 => k8s.io/kubectl v0.0.0-20201113183129-6c3805304969
-	k8s.io/kubelet v0.0.0 => k8s.io/kubelet v0.0.0-20201113180421-609649667e1d
-	k8s.io/legacy-cloud-providers v0.0.0 => k8s.io/legacy-cloud-providers v0.0.0-20201113182353-b77454164b5d
-	k8s.io/metrics v0.0.0 => k8s.io/metrics v0.0.0-20201113175240-4a8e322202a4
-	k8s.io/sample-apiserver v0.0.0 => k8s.io/sample-apiserver v0.0.0-20201113174004-1f754d3b7d52
+	k8s.io/api v0.0.0 => k8s.io/api v0.0.0-20210408192244-5d0d5b56b511
```
Comment (marquiz) [inline go.mod:41]:
```suggestion
	k8s.io/api => k8s.io/api v0.21.0
```

and ditto X times below...

**Review by marquiz — COMMENTED** [review]
Should we update directly to v1.22.2?

**Comment by klihub** [comment]
> Should we update directly to v1.22.2?

Sure, if we can find a working combo of the pseudo-sub-repos split out from kubernetes staging. To me it looks like the highest commonly tagged release version is v0.21.0 (so v1.21.0 in your notation).

**Comment by marquiz** [comment]
Hmm, why not bump to v1.22 directly? #722

**Comment by klihub** [comment]
> Hmm, why not bump to v1.22 directly? #722

Argh... because apparently I fscked up something and failed to find the proper tags in the repos.

**Pull request closed without merging**